app/webpacker/gtm.js

```diff
@@ -0,0 +1,55 @@
+import { CookiePreferences } from 'cookie_preferences'
+
+export default class Gtm {
+  constructor(id, nonce) {
+    this.id = id
+    this.nonce = nonce
+  }
+
+  init() {
+    this.initWindow()
+    this.sendDefaultConsent()
+    this.initContainer()
+  }
+
+  initWindow() {
+    window.dataLayer = window.dataLayer || []
+
+    function gtag() {
```

Comment:
Wondering what this nested function is doing? I don't think it's called anywhere?

Reply:
And arguments will always be null? Might be just that I don't understand whats going on

Reply:
So we set it to `window.gtag` a few lines below and then its called further down when we send consent to GTM:

```
window.gtag('consent', 'default', this.consent())
```

It's because the consent sends before GTM is initialised (and GTM would otherwise define `window.gtm` itself) - we could call the `dataLayer` directly but this was the recommended way of doing it, I think just so it goes through the `gtag` interface like other calls would after GTM has loaded 🤷‍♂️

Reply:
`arguments` is a keyword that will return whatever gets sent into the `gtag` method ^

Reply:
Ah, yeah I see. It's me being stupid.  Forgot you can just assign functions to variables like that. Used to seeing `const something = () => {}` rather than `function something () {}; const otherThing = something`